packages/backend/index.js

```diff
@@ -70,6 +70,86 @@ app.get("/flocks/:code/basket", (req, res) => {
 	res.send(`Options of flock ${req.params.code}`);
 });
 
+app.get("/flocks/:coop_name/:chick/vote", async (req, res) => {
+	console.log(`GET /flocks/${req.params.coop_name}/${req.params.chick}/vote`);
+	const coopName = req.params.coop_name;
+	const chickName = req.params.chick;
+
+	// check if flock and chick exists
+	const flock = await findFlockByCode(coopName);
+	if (!flock) {
+		res.status(404).send({ message: "Flock not found" });
+		return;
+	}
+	const chick = flock.chicks.find((chick) => chick.name === chickName);
+	if (!chick) {
+		res.status(404).send({ message: "Chick not found" });
+		return;
+	}
+
+	// get a restaurant that hasn't been voted on yet
+	const existingVotes = chick.preferences.map((preference) => preference.egg.toString());
+	const remainingOptions = flock.basket.filter(
+		(egg) => !existingVotes.includes(egg._id.toString())
+	);
+	// console.log("remainingOptions", remainingOptions);
+
+	if (remainingOptions.length === 0) {
+		res.status(204).send();
+		return;
+	}
+
+	// return a random restaurant
```

Comment:
are you intending to return a random restaurant from the list of restaurants or the first restaurant on the list?